source/src/js/carousel.js

```diff
@@ -0,0 +1,195 @@
+/**
+ * @param {number} number
+ * @param {number} mod
+ * @return {number}
+ */
+function modulo(number, mod) {
+  let result = number % mod;
+  if (result < 0) {
+    result += mod;
+  }
+  return result;
+}
```

Comment:
```suggestion
const mathMod = (n, d) => ((n % d) + d) % d;
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Remainder#:~:text=To%20obtain%20a%20modulo%20in%20JavaScript%2C%20in%20place%20of%20n%20%25%20d%2C%20use%20((n%20%25%20d)%20%2B%20d)%20%25%20d.